### ISSUE TITLE
Incorrect lint warnings when `ol-prefix` style set to `"ordered"`.

### ISSUE THREAD

**Issue by nbird11** [DavidAnson/vscode-markdownlint#374]
# See title

Seems like all of the ol-prefix style options are broken, not just `"ordered"`

## What did you do?

- In a markdown file, have ordered list items prefixed with 1., 2., 3., etc. rather than 1., 1., 1., etc.

```md
...
## Example

1. Title One

- Lorem ipsum
- Lorem ipsum

2. Title Two

- Lorem ipsum
- Lorem ipsum

3. Title Three

- Lorem ipsum
- Lorem ipsum
```

- In vscode setting.json, set markdownlint.config ol-prefix setting to style ordered:

```json
    "markdownlint.config": {
        "ol-prefix": {
            "style": "ordered"
        }
    }
```

## What did you expect to happen?

- Expected for my ordered list numbering not to be highliighted as a warning.

## What actually happened?

- Each ordered list prefix that is not `1.` get's the yellow lint warning underline and shows up in the `Problems` tab with the below message.

## What messages or errors were there?

- `MD029/ol-prefix: Ordered list item prefix [Expected: 1; Actual: 2; Style: 1/2/3] markdownlint[MD029](https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md029.md)`

## How can the issue be reproduced?

- See above.

## What version were you using?

Installation

|  |  | 
| --- | --- |
| Identifier | davidanson.vscode-markdownlint |
| Version | 0.59.0 |
| Last Updated | 2025-02-18, 20:01:54 |

## What operating system were you using?

- Windows

**Comment by nschonni** [comment]
The unordered lists inside aren't indented enough/at all, so parsers will stop the list and create a new OL for each number

**Comment by DavidAnson** [comment]
Agree. The example shows 3 separate ordered lists, so the lint message is correct.
